buildSrc/src/main/groovy/io/micronaut/guides/GuidesPlugin.groovy

```diff
@@ -390,4 +432,13 @@ class GuidesPlugin implements Plugin<Project> {
     private static String quote(it) {
         '"' + it + '"'
     }
+
+    private static boolean isGraalVMJava() {
+        return Arrays.asList("jvmci.Compiler", "java.vendor.version", "java.vendor")
+                .stream()
+                .anyMatch(propertyName -> {
+                    String value = System.getProperty(propertyName)
+                    return value != null && value.toLowerCase(Locale.ENGLISH).contains("graal");
+                })
+    }
```

Comment:
We had problems with this way of detecting Graal, as there's one version (only on CI) that didn't fill any of these properties 

See https://github.com/micronaut-projects/micronaut-kotlin/issues/558#issuecomment-1715380535

And https://discord.com/channels/1121511613250412714/1146748986787647639/1147113109031505951

However, the downside will be that these tests don't run, not a failure of the pipeline

Reply:
it works for me with 

```
% java --version
java 17.0.8 2023-07-18 LTS
Java(TM) SE Runtime Environment Oracle GraalVM 17.0.8+9.1 (build 17.0.8+9-LTS-jvmci-23.0-b14)
Java HotSpot(TM) 64-Bit Server VM Oracle GraalVM 17.0.8+9.1 (build 17.0.8+9-LTS-jvmci-23.0-b14, mixed mode, sharing)
```

Reply:
Yes, the problem was it didn't work on the JDK installed by CI

Worked fine locally